homeassistant/components/recorder/migration.py

```diff
@@ -1040,7 +1070,12 @@ class _SchemaVersion2Migrator(_SchemaVersionMigrator, target_version=2):
     def _apply_update(self) -> None:
         """Version specific update method."""
         # Create compound start/end index for recorder_runs
-        _create_index(self.session_maker, "recorder_runs", "ix_recorder_runs_start_end")
+        _create_index(
+            self.instance,
+            self.session_maker,
```

Comment:
Probably for a future PR: Do we ever need to pass session_maker if we pass `instance`?

Reply:
I followed this pattern because we pass the recorder `instance` and `session_maker` separately when calling migration functions from `recorder/core.py`:
https://github.com/home-assistant/core/blob/f10c60bbf6bdaa6a8517f20a154977ed0d227529/homeassistant/components/recorder/core.py#L960-L966

https://github.com/home-assistant/core/blob/f10c60bbf6bdaa6a8517f20a154977ed0d227529/homeassistant/components/recorder/core.py#L924

Cleaning it up seems like a good idea, but in a separate PR then.